milli/src/search/new/ranking_rule_graph/fid/mod.rs

```diff
@@ -80,6 +81,35 @@ impl RankingRuleGraphTrait for FidGraph {
             ));
         }
 
+        // always lookup the max_fid if we don't already and add an artificial condition for max scoring
+        let max_fid: Option<u16> = {
+            if let Some(max_fid) = ctx
+                .index
+                .searchable_fields_ids(ctx.txn)?
+                .map(|field_ids| field_ids.into_iter().max())
+            {
+                max_fid
+            } else {
+                ctx.index.fields_ids_map(ctx.txn)?.ids().max()
+            }
+        };
```

Comment:
I was ready to write a comment to tell you to use `fields_ids_map` in case `searchable_fields_ids` was `None`, but you did it already :D That's a detail of the codebase that is too tricky IMO.

Reply:
I agree IMO as a caller in `fid` I shouldn't have to worry about this details.

A possible fix could be to change the `searchable_fields_ids` method to return an enum:

```rust
enum SearchableFieldIds {
    UserDefined(Vec<FieldId>),
    Inferred(Vec<FieldId>),
}

impl SearchableFieldIds {
    pub fn into_vec(self) -> Vec<FieldId> { /* ... */ }
}
```

Then my code could become:

```rust
let max_fid = ctx.index.searchable_fields_ids(ctx.txn)?.into_iter().max();
```

while still leaving the possibility for interested callers to match on the enum to know if the searchable fields are inferred or user-defined.

Reply:
That would be great :) but not strictly needed to do it in this PR IMO